File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/updateconfigs/UpdateConfigsImplV0_6.java
package com.evernym.verity.sdk.protocols.updateconfigs;

import com.evernym.verity.sdk.exceptions.VerityException;
import com.evernym.verity.sdk.protocols.AbstractProtocol;
import com.evernym.verity.sdk.protocols.updateconfigs.v0_6.UpdateConfigsV0_6;
import com.evernym.verity.sdk.utils.Context;
import org.json.JSONArray;
import org.json.JSONObject;

import java.io.IOException;

/*
 * NON_VISIBLE
 *
 * This is an implementation of UpdateConfigsImplV0_6 but is not viable to user of Verity SDK. Created using the
 * static UpdateConfigs class
 */
class UpdateConfigsImplV0_6 extends AbstractProtocol implements UpdateConfigsV0_6 {

    final String UPDATE_CONFIGS = "update";
    final String GET_STATUS = "get-status";

    String name;
    String logoUrl;

    UpdateConfigsImplV0_6(String name, String logoUrl) {
        super();
        this.name = name;
        this.logoUrl = logoUrl;
    }

    UpdateConfigsImplV0_6() {
        super();
    }

    @Override
    public void update(Context context) throws IOException, VerityException {
        send(context, updateMsg(context));
    }

    @Override
    public JSONObject updateMsg(Context context) {
        JSONObject message = new JSONObject();
        message.put("@type", messageType(UPDATE_CONFIGS));
        message.put("@id", getNewId());
        JSONArray configs = new JSONArray();
        JSONObject item1 = new JSONObject();
        item1.put("name", "name");
        item1.put("value", this.name);
        configs.put(item1);
        JSONObject item2 = new JSONObject();
        item2.put("name", "logoUrl");
        item2.put("value", this.logoUrl);
        configs.put(item2);
        message.put("configs", configs);
        return message;
    }

    @Override
    public byte[] updateMsgPacked(Context context) throws VerityException {
        return packMsg(context, updateMsg(context));
    }

    public void status(Context context) throws IOException, VerityException {
        send(context, statusMsg(context));
    }

    @Override
    public JSONObject statusMsg(Context context) {
        JSONObject msg = new JSONObject();
        msg.put("@type", messageType(GET_STATUS));
        msg.put("@id", getNewId());
        addThread(msg);

        return msg;
    }

    @Override
    public byte[] statusMsgPacked(Context context) throws VerityException {
        return packMsg(context, statusMsg(context));
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/UpdateConfigs/UpdateConfigsImplV0_6.cs
using System;
using System.Json;

namespace VeritySDK
{
    /// <summary>
    /// This is an implementation of UpdateConfigsImplV0_6 but is not viable to user of Verity SDK.
    /// </summary>
    public class UpdateConfigsImplV0_6 : UpdateConfigsV0_6
    {
        string UPDATE_CONFIGS = "update";
        string GET_STATUS = "get-status";

        string name;
        string logoUrl;

        public UpdateConfigsImplV0_6() { }

        public UpdateConfigsImplV0_6(string name, string logoUrl)
        {
            this.name = name;
            this.logoUrl = logoUrl;
        }

        public override void update(Context context)
        {
            send(context, updateMsg(context));
        }

        public override JsonObject updateMsg(Context context)
        {
            JsonObject message = new JsonObject();
            message.Add("@type", messageType(UPDATE_CONFIGS));
            message.Add("@id", getNewId());
            JsonArray configs = new JsonArray();
            JsonObject item1 = new JsonObject();
            item1.Add("name", "name");
            item1.Add("value", this.name);
            configs.Add(item1);
            JsonObject item2 = new JsonObject();
            item2.Add("name", "logoUrl");
            item2.Add("value", this.logoUrl);
            configs.Add(item2);
            message.Add("configs", configs);
            return message;
        }

        public override byte[] updateMsgPacked(Context context)
        {
            return packMsg(context, updateMsg(context));
        }

        public override void status(Context context)
        {
            send(context, statusMsg(context));
        }

        public override JsonObject statusMsg(Context context)
        {
            JsonObject msg = new JsonObject();
            msg.Add("@type", messageType(GET_STATUS));
            msg.Add("@id", getNewId());
            addThread(msg);

            return msg;
        }

        public override byte[] statusMsgPacked(Context context)
        {
            return packMsg(context, statusMsg(context));
        }
    }
}